src/services/gateway.service.ts

```diff
@@ -584,8 +586,16 @@ function setDefaultChain(appId: number, data: any, network: Network) {
 }
 
 async function getChainIDUsingRPCUrl(rpcURL: string) {
-  const payload = { jsonrpc: '2.0', method: 'eth_chainId', params: [], id: 1 } // id - dummy value
-  return axios.post(rpcURL, payload)
+  const provider: providers.WebSocketProvider | providers.JsonRpcProvider =
+    isValidWsUrl(rpcURL)
+      ? new providers.WebSocketProvider(rpcURL)
+      : new providers.JsonRpcProvider(rpcURL)
+  const chainId = await provider.getNetwork().then((network) => network.chainId)
```

Comment:
If you are using await then why use `.then`?
```suggestion
  const network = await provider.getNetwork()
```

Reply:
Fixed @mmjee